gtk4/src/subclass/accessible_text.rs

```diff
@@ -0,0 +1,533 @@
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface = type_data.as_ref().parent_interface::<AccessibleText>()
+                as *const ffi::GtkAccessibleTextInterface;
+
+            let func = (*parent_iface)
+                .get_attributes
+                .expect("no parent \"get_attributes\" implementation");
+
+            let mut n_ranges = MaybeUninit::uninit();
+            let mut ranges = std::ptr::null_mut();
+            let mut attribute_names = std::ptr::null_mut();
+            let mut attribute_values = std::ptr::null_mut();
+
+            let is_set: bool = from_glib(func(
+                self.obj()
+                    .unsafe_cast_ref::<AccessibleText>()
+                    .to_glib_none()
+                    .0,
+                offset,
+                n_ranges.as_mut_ptr(),
+                &mut ranges,
+                &mut attribute_names,
+                &mut attribute_values,
+            ));
+
+            if !is_set
+                || n_ranges.assume_init() == 0
+                || ranges.is_null()
+                || attribute_names.is_null()
+                || attribute_values.is_null()
+            {
+                Vec::new()
+            } else {
+                let mut names = glib::StrV::from_glib_full(attribute_names).into_iter();
+                let mut values = glib::StrV::from_glib_full(attribute_values).into_iter();
+
+                glib::Slice::from_glib_container_num(ranges, n_ranges.assume_init())
+                    .into_iter()
+                    .flat_map(|range| {
+                        if let (Some(name), Some(value)) = (names.next(), values.next()) {
+                            Some((range, name, value))
+                        } else {
+                            None
+                        }
+                    })
+                    .collect()
+            }
+        }
+    }
+
+    fn parent_caret_position(&self) -> u32 {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface = type_data.as_ref().parent_interface::<AccessibleText>()
+                as *const ffi::GtkAccessibleTextInterface;
+
+            let func = (*parent_iface)
+                .get_caret_position
+                .expect("no parent \"get_caret_position\" implementation");
+
+            func(
+                self.obj()
+                    .unsafe_cast_ref::<AccessibleText>()
+                    .to_glib_none()
+                    .0,
+            )
+        }
+    }
+
+    fn parent_contents(&self, start: u32, end: u32) -> Option<glib::Bytes> {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface = type_data.as_ref().parent_interface::<AccessibleText>()
+                as *const ffi::GtkAccessibleTextInterface;
+
+            let func = (*parent_iface)
+                .get_contents
+                .expect("no parent \"get_contents\" implementation");
```

Comment:
Is this required? Usually we return None if there is no override/default implementation